hazelcast/src/test/java/com/hazelcast/test/AbstractHazelcastClassRunner.java

```diff
@@ -353,6 +356,28 @@ public void evaluate() throws Throwable {
         };
     }
 
+    private void detectThreadLeak() {
+        boolean threadLeak = false;
+        Set<Thread> threads = Thread.getAllStackTraces().keySet();
+        StringBuilder leakingThreads = new StringBuilder();
+
+        for (Thread thread : threads) {
+            if (thread.getName().toLowerCase().contains("hz") && thread.isAlive()) {
```

Comment:
Berzerk mode: on :smile: 
We cannot add the usual `hazelcastInstance` string to the match string, since WAN tests using mocked network add cluster-specific UUID to the instance names. An example test name from such a WAN test: `hz.confB-2e985098-6450-4c4c-b94c-0f965f85cd58-01.cached.thread-2`.

Reply:
As I recall, this was a quick solution to the issue that instance names must be unique within the JVM AFAIK when running tests in parallel. Any other JVM-unique name will do as well, the WAN tests don't rely on this specific format.

Reply:
https://github.com/hazelcast/hazelcast/blob/25b0dd92aeb9d5c35b45a8084dbad455c0dc9f6a/hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceFactory.java#L59

Reply:
Yes, maybe we should find a more strategic naming convention for the parallel WAN tests before continuing this.